ui/main/src/app/modules/monitoringexp/monitoringexp.component.ts

```diff
@@ -0,0 +1,315 @@
+        if (this.listOfProcessesForRequest.length && !filtersMap.has('process'))
+            filters.push(new FilterModel('process', null, FilterMatchTypeEnum.IN, this.listOfProcessesForRequest));
+
+        this.columnFilters.forEach(filter => filters.push(filter));
+
+        const selectedFields: string[] = [];
+        this.processMonitoring.forEach(column => {
+            selectedFields.push(column.field);
+        });
+
+        return new CardsFilter(page, size, isAdminMode, true, false, filters, selectedFields);
+    }
+
+    cardPostProcessing(card) {
+        const isThirdPartyPublisher = card.publisherType === 'EXTERNAL';
+        const sender = isThirdPartyPublisher ? card.publisher : this.entitiesService.getEntityName(card.publisher);
+
+        let representative = '';
+        if (card.representativeType && card.representative) {
+            const isThirdPartyRepresentative = card.representativeType === 'EXTERNAL';
+            representative = isThirdPartyRepresentative
+                ? card.representative
+                : this.entitiesService.getEntityName(card.representative);
+        }
+        card.sender = sender;
+        card.representative = representative;
+
+        card.processName = this.processNames.get(card.process);
+        card.stateColor = this.stateColors.get(card.process + '.' + card.state);
+        card.severityNumber = this.mapSeverity.get(card.severity.toLocaleLowerCase());
+    }
+
+    search() {
+        this.firstQueryHasBeenDone = false;
+        this.columnFilters = [];
+        this.sendFilterQuery(0);
+        this.page = 1;
+    }
+
+    onPageChange(currentPage): void {
+        // page on ngb-pagination component starts at 1, and page on backend starts at 0
+        this.sendFilterQuery(currentPage - 1);
+        this.page = currentPage;
+    }
+
+    onTableFilterChange(filterModel) {
+        this.columnFilters = [];
+        Object.keys(filterModel).forEach(column => {
+            const type : string = filterModel[column].type;
+            this.columnFilters.push(new FilterModel(column, filterModel[column].filterType, FilterMatchTypeEnum[type.toUpperCase()], [filterModel[column].filter]))
+        });
+        this.sendFilterQuery(0);
+    }
+
+    displayTime(date) {
+        return this.dateTimeFormatter.getFormattedDateAndTimeFromEpochDate(date);
+    }
+
+    exportToExcel(): void {
+        const exportArchiveData = [];
+
+        const modalOptions: NgbModalOptions = {
+            centered: true,
+            backdrop: 'static', // Modal shouldn't close even if we click outside it
+            size: 'sm'
+        };
+        this.modalRef = this.modalService.open(this.exportTemplate, modalOptions);
+
+        const filter = this.getFilter(0, this.resultsNumber, this.filtersTemplate.filters);
+
+        this.cardService
+            .fetchFilteredCards(filter)
+            .pipe(takeUntil(this.unsubscribe$))
```

Comment:
You do not need to unsubscribe as the observable is close automatically in this case

Reply:
Done. Removed.